File: clients/python/model/bullet_params.py
class BulletParams:
    def __init__(self, speed, size, damage):
        self.speed = speed
        self.size = size
        self.damage = damage
    @staticmethod
    def read_from(stream):
        speed = stream.read_double()
        size = stream.read_double()
        damage = stream.read_int()
        return BulletParams(speed, size, damage)
    def write_to(self, stream):
        stream.write_double(self.speed)
        stream.write_double(self.size)
        stream.write_int(self.damage)
    def __repr__(self):
        return "BulletParams(" + \
            repr(self.speed) + "," + \
            repr(self.size) + "," + \
            repr(self.damage) + \
            ")"

File: clients/ruby/model/bullet_params.rb
class BulletParams
    attr_accessor :speed
    attr_accessor :size
    attr_accessor :damage
    def initialize(speed, size, damage)
        @speed = speed
        @size = size
        @damage = damage
    end
    def self.read_from(stream)
        speed = stream.read_double()
        size = stream.read_double()
        damage = stream.read_int()
        BulletParams.new(speed, size, damage)
    end
    def write_to(stream)
        stream.write_double(@speed)
        stream.write_double(@size)
        stream.write_int(@damage)
    end
end
